client/src/components/Game/GameInfo/GameInfo.tsx

```diff
@@ -14,6 +34,117 @@ const GameInfo = ({ socket, exitCallback }: { socket: Socket; exitCallback: Func
         }
     };
 
+    useEffect(() => {
+        socket.on('connect', () => {
+            setTimePeriod({
+                name: 'Loading',
+                toolTip: 'Getting Data From Server',
+                maxDuration: -1,
+            } as TimePeriodAndDuration);
+
+            // setTimeout(() => {
+            //     setTimePeriod({
+            //         name: 'Pre-Game',
+            //         toolTip: 'Waiting for host to start lobby',
+            //         maxDuration: 10,
+            //     });
+            //     setTimeRemaining(10);
+            // }, 1000);
+        });
+        socket.on(
+            'timePeriodUpdate',
+            (
+                {
+                    name,
+                    description,
+                    durationSeconds,
+                }: {
+                    name: string;
+                    description: string;
+                    durationSeconds: number;
+                },
+                timeLeft: number,
+            ) => {
+                setTimePeriod({
+                    name,
+                    toolTip: description,
+                    maxDuration: timeLeft,
+                });
+                setTimeRemaining(durationSeconds);
+            },
+        );
+        socket.on('disconnect', () => {
+            setTimePeriod({
+                name: `${timePeriod.name} (Disconnected)`,
+                toolTip: 'Disconnected From The Server',
+                maxDuration: -1,
+            } as TimePeriodAndDuration);
+        });
+        return () => {
+            socket.off('timePeriodUpdate');
+            // socket.off('connect');
+            // socket.off('disconnect');
+        };
+    }, [socket, timePeriod.name]);
+
+    // decrementing time
+    useEffect(() => {
+        const myInterval = setInterval(() => {
+            if (timeRemaining > 0) {
+                setTimeRemaining(timeRemaining - 1 * (UPDATE_INTERVAL / 1000));
+            }
+        }, UPDATE_INTERVAL);
+        return () => {
+            clearInterval(myInterval);
+        };
+    });
```

Comment:
Maybe explicitly add a dependency of [UPDATE_INTERVAL] here, just in case something in the future causes a re-render outside of your update interval. This way, the `useEffect()` here will do exactly what you've asked it to by itself instead of having it rely on a different module/function which might lead to spaghetti code (idk what I'm doing lmao)

Reply:
This is actually a good idea since the eventual plan is to be able to configure `UPDATE_INTERVAL` in some sort of settings menu.

Reply:
This is actually a good idea since the eventual plan is to be able to configure `UPDATE_INTERVAL` in some sort of settings menu.